src/frontend/mosh-server.cc

```diff
@@ -551,6 +576,8 @@ void serve( int host_fd, Terminal::Complete &terminal, ServerConnection &network
 	sel.add_fd( host_fd );
       }
 
+      network.oob()->pre_poll();
```

Comment:
Having all these calls to network.oob, and the pre/post machinery, feels awkward to me.  It seems to me that this is all a workaround for having `network`'s `select` calls here at top level.  Perhaps we can push the selects down into `Network::Transport`, and have that select for and call out to the OOB object (or maybe even push the select object down another level to OOB as well).

Adding another set of data flows to Mosh is going to be a little awkward no matter how it's done, but I think it can be a little less awkward than this :)